### PR TITLE
지역번호 개발기능 추가

### PR DESCRIPTION
**Pull request by ska202**
지역번호 개발기능 추가

### CODE REVIEW

**Comment by ska202** [comment]
### :construction: Codebot Build Result

result is failure

>[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.115 s
[INFO] Finished at: 2020-05-17T05:02:11Z
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is failure

>[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.088 s
[INFO] Finished at: 2020-05-17T05:09:41Z
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is success

>[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.462 s
[INFO] Finished at: 2020-05-17T05:12:32Z
[INFO] ------------------------------------------------------------------------

**Comment by CODARI-JAMSIL** [comment]
### :sparkles: Codebot JUnit test results

Result is failure

>BUILD FAILURE<BR>[INFO] ------------------------------------------------------------------------<BR>[INFO] Total time:  22.616 s<BR>[INFO] Finished at: 2020-05-17T05:13:00Z<BR>[INFO] ------------------------------------------------------------------------<BR>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project spring-framework-petclinic: There are test failures.<BR>[ERROR] <BR>[ERROR] Please refer to /data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev/target/surefire-reports for the individual test results.<BR>[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.<BR>[ERROR] -> [Help 1]<BR>[ERROR] <BR>[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.<BR>[ERROR] Re-run Maven using the -X switch to enable full debug logging.<BR>[ERROR] <BR>[ERROR] For more information about the errors and possible solutions, please read the following articles:<BR>[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException<BR>

**Comment by CODARI-JAMSIL** [comment]
SAM analyzing failed

**Comment by CODARI-JAMSIL** [comment]
:checkered_flag:  Code Spell Check Result

No Wrong Word Detected - Good Job :+1:

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is success

>[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.427 s
[INFO] Finished at: 2020-05-17T05:16:06Z
[INFO] ------------------------------------------------------------------------

**Comment by CODARI-JAMSIL** [comment]
### :sparkles: Codebot JUnit test results

Result is failure

>BUILD FAILURE<BR>[INFO] ------------------------------------------------------------------------<BR>[INFO] Total time:  22.429 s<BR>[INFO] Finished at: 2020-05-17T05:16:34Z<BR>[INFO] ------------------------------------------------------------------------<BR>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project spring-framework-petclinic: There are test failures.<BR>[ERROR] <BR>[ERROR] Please refer to /data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev/target/surefire-reports for the individual test results.<BR>[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.<BR>[ERROR] -> [Help 1]<BR>[ERROR] <BR>[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.<BR>[ERROR] Re-run Maven using the -X switch to enable full debug logging.<BR>[ERROR] <BR>[ERROR] For more information about the errors and possible solutions, please read the following articles:<BR>[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException<BR>

**Comment by CODARI-JAMSIL** [comment]
SAM analyzing failed

**Comment by CODARI-JAMSIL** [comment]
:checkered_flag:  Code Spell Check Result

No Wrong Word Detected - Good Job :+1:

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is success

>[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.417 s
[INFO] Finished at: 2020-05-17T05:17:53Z
[INFO] ------------------------------------------------------------------------

**Comment by CODARI-JAMSIL** [comment]
### :sparkles: Codebot JUnit test results

Result is failure

>BUILD FAILURE<BR>[INFO] ------------------------------------------------------------------------<BR>[INFO] Total time:  22.463 s<BR>[INFO] Finished at: 2020-05-17T05:18:21Z<BR>[INFO] ------------------------------------------------------------------------<BR>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project spring-framework-petclinic: There are test failures.<BR>[ERROR] <BR>[ERROR] Please refer to /data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev/target/surefire-reports for the individual test results.<BR>[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.<BR>[ERROR] -> [Help 1]<BR>[ERROR] <BR>[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.<BR>[ERROR] Re-run Maven using the -X switch to enable full debug logging.<BR>[ERROR] <BR>[ERROR] For more information about the errors and possible solutions, please read the following articles:<BR>[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException<BR>

**Comment by CODARI-JAMSIL** [comment]
SAM analyzing failed

**Comment by CODARI-JAMSIL** [comment]
:checkered_flag:  Code Spell Check Result

No Wrong Word Detected - Good Job :+1:

File: src/main/java/org/springframework/samples/petclinic/web/OwnerController.java [hunk 1]
```diff
@@ -76,31 +76,74 @@ public String initFindForm(Map<String, Object> model) {
...
+				}
+			}
+		}
+		model.put("selections", results);
+		return "owners/ownersList";
+	}
+}
 
+private boolean isInternetTel(String initNum) {
+	return initNum == "070";
```
Comment (CODARI-JAMSIL) [inline src/main/java/org/springframework/samples/petclinic/web/OwnerController.java:145]: MAJOR Comparison of String parameter using == or != in org.springframework.samples.petclinic.web.OwnerController.isInternetTel(String) [rule](http://10.0.10.10/sonar/coding_rules#rule_key=findbugs%3AES_COMPARING_PARAMETER_STRING_WITH_EQ)

**Comment by CODARI-JAMSIL** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is success

>[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.408 s
[INFO] Finished at: 2020-05-17T05:27:11Z
[INFO] ------------------------------------------------------------------------

**Comment by CODARI-JAMSIL** [comment]
### :sparkles: Codebot JUnit test results

Result is failure

>BUILD FAILURE<BR>[INFO] ------------------------------------------------------------------------<BR>[INFO] Total time:  22.709 s<BR>[INFO] Finished at: 2020-05-17T05:28:27Z<BR>[INFO] ------------------------------------------------------------------------<BR>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project spring-framework-petclinic: There are test failures.<BR>[ERROR] <BR>[ERROR] Please refer to /data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev/target/surefire-reports for the individual test results.<BR>[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.<BR>[ERROR] -> [Help 1]<BR>[ERROR] <BR>[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.<BR>[ERROR] Re-run Maven using the -X switch to enable full debug logging.<BR>[ERROR] <BR>[ERROR] For more information about the errors and possible solutions, please read the following articles:<BR>[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException<BR>

**Comment by CODARI-JAMSIL** [comment]
SAM analyzing failed

**Comment by CODARI-JAMSIL** [comment]
:checkered_flag:  Code Spell Check Result

No Wrong Word Detected - Good Job :+1:

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is success

>[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.594 s
[INFO] Finished at: 2020-05-17T05:38:02Z
[INFO] ------------------------------------------------------------------------

**Comment by CODARI-JAMSIL** [comment]
### :sparkles: Codebot JUnit test results

Result is failure

>BUILD FAILURE<BR>[INFO] ------------------------------------------------------------------------<BR>[INFO] Total time:  22.985 s<BR>[INFO] Finished at: 2020-05-17T05:39:21Z<BR>[INFO] ------------------------------------------------------------------------<BR>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project spring-framework-petclinic: There are test failures.<BR>[ERROR] <BR>[ERROR] Please refer to /data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev/target/surefire-reports for the individual test results.<BR>[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.<BR>[ERROR] -> [Help 1]<BR>[ERROR] <BR>[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.<BR>[ERROR] Re-run Maven using the -X switch to enable full debug logging.<BR>[ERROR] <BR>[ERROR] For more information about the errors and possible solutions, please read the following articles:<BR>[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException<BR>

**Comment by CODARI-JAMSIL** [comment]
### :construction: Codebot Build Result

result is success

>[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.544 s
[INFO] Finished at: 2020-05-17T05:42:06Z
[INFO] ------------------------------------------------------------------------

**Comment by CODARI-JAMSIL** [comment]
### :sparkles: Codebot JUnit test results

Result is failure

>BUILD FAILURE<BR>[INFO] ------------------------------------------------------------------------<BR>[INFO] Total time:  23.000 s<BR>[INFO] Finished at: 2020-05-17T05:43:23Z<BR>[INFO] ------------------------------------------------------------------------<BR>[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project spring-framework-petclinic: There are test failures.<BR>[ERROR] <BR>[ERROR] Please refer to /data/codebot/workspace/CodeBotAi/spring-framework-petclinic/dev/target/surefire-reports for the individual test results.<BR>[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.<BR>[ERROR] -> [Help 1]<BR>[ERROR] <BR>[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.<BR>[ERROR] Re-run Maven using the -X switch to enable full debug logging.<BR>[ERROR] <BR>[ERROR] For more information about the errors and possible solutions, please read the following articles:<BR>[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException<BR>

**Comment by CODARI-JAMSIL** [comment]
### :page_with_curl: SAM(Software Architecture Maturity) Result
  ** SAM Score is Total Repository Source Code analysis results
  ** Detail Contents are only about this pull request.

| Item | Score | Detail & Guide |
| :----: | :----: | :---- |
| CC<br>(Cyclomatic Complexity) | 4.51 | ▶ 유지보수를 어렵게 하는 [소스 복잡도(CC)](http://devops.sdsdev.co.kr/confluence/display/SPC/Complexity)는 적정하게 관리되고 있습니다.<br>다만, 변경 소스 중 복잡도가 23(기준 21)인 메소드가 존재 하여(하단 목록 참조) 개선이 필요합니다.<br>**[개선 가이드]**<br> 1) 메소드 내부의 분리 가능한 부분을 [메소드로 추출(Extract Method)](http://devops.sdsdev.co.kr/confluence//pages/viewpage.action?pageId=97897028) <br> 2) [불필요한 로직 삭제(Substitute Algorithm)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=97897019) <br> 3) 조건문이 복잡한 경우의 [조건문 간결화(Simplify Conditions)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=97897043) 등의 방법이 있습니다. |
| DC<br>(Duplicated Code) | 4.80 | ▶재사용을 어렵게 하고, 코드 개선 노력을 증가시는 [중복 코드(DC)](http://devops.sdsdev.co.kr/confluence/display/SPC/Duplications)는 매우 적절히 관리되고 있습니다.<br> |
| CBO<br>(Coupling Between Objects) | 5.00 | ▶다른 모듈에 의존하는 정도를 나타내는 [결합도(CBO)](http://devops.sdsdev.co.kr/confluence/display/SPC/Coupling)는 매우 적절히 관리되고 있습니다. |
| LOC<br>(Lines of Code) | 5.00 | ▶ 단일 역할을 수행해야 하는 하나의 소스 크기를 측정하는 [소스당 코드규모(LOC)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=99559241)는 매우 적절히 관리되고 있습니다. |
| ND<br>(Normalized Distance) | 0.00 | ▶ 패키지 변경영향을 평가하는 [패키지안정성(ND)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=99563012)은 매우 높고, 전체적으로 개선이 필요합니다.<br>패키지안정성을 평가하는 ND가 높으면 패키지 외부 변경에 영향을 많이 받거나, 패키지 내부의 수정이 외부에 영향을 주기 때문에 유지보수를 어렵게 합니다.<br>변경된 소스는 아니지만, 전체 패키지 중 개선이 필요한 패키지가 있습니다.<br>**[개선 가이드]**<br> 1) [인터페이스 도출(Extract Interface)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=97896896) 을 통해 패키지에 대한 추상화를 하거나<br>  2) [객체간의 기능을 적정하게 이동(Moving features between objects)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=97897029) 하여 불필요한 상호 참조를 없애는 등의 리팩토링 방법이 있습니다. |
| COC<br>(Cohesion of Class) | 0.50 | ▶ 클래스에 대한 단일 책임 원칙(Single Responsibility Principle)을 평가하는 [응집도(Cohesion)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=99563012)는 높으며, 전체적으로 개선이 필요합니다. 응집도가 낮으면 여러 역할을 가지고 있기 때문에 이해 및 재사용이 어려울뿐만 아니라, 기능 개선에 따른 변경의 영향을 높습니다.<br>변경된 소스는 아니지만, 전체 중 개선이 필요한 소스가 있습니다.<br>**[개선 가이드]**<br> 1) [객체간의 기능을 적정하게 이동(Moving features between objects)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=97897029) 을 통해 관련 있는 기능과 속성만을 묶어주거나,<br>  2) [데이터 체계화(Organizing Data)](http://devops.sdsdev.co.kr/confluence/pages/viewpage.action?pageId=97897041) 를 통해 데이터를 좀 더 쉽게 다룰 수 있는 등의 리팩토링 방법이 있습니다. |

  ※ SCORE Range 

| Score Code & Name | Score Range |
| :---- | :---- |
| EX (Excellent) | 5.0 |
| ST (Statisfied) | 4.0~4.99 |
| SR (Some Refactoring  needed) | 3.0~3.99 |
| OR (Overall  Refactoring needed) | 2.0~2.99 |
| RD (Consider  Redevelopment) | 0.0~1.99 |

### [Need Improved Code Lists in PR]
| Item | Path | Remarks |
| :----: | :---- | :---- |
| CC | src/main/java/org/springframework/samples/petclinic/web/OwnerController.java | Line: 80<br>Method Name: processFindForm<br>Complexity: 23 |

**Comment by CODARI-JAMSIL** [comment]
:checkered_flag:  Code Spell Check Result

No Wrong Word Detected - Good Job :+1:

**Pull request closed without merging**